### PR TITLE
7 -  libraries:iio: Generate xml from iio_interface 

### PR DESCRIPTION
**Pull request by mchindri**
Built on top of: https://github.com/analogdevicesinc/no-OS/pull/670

Last 2 commits relevant

### CODE REVIEW

**Comment by mchindri** [comment]
v2: 
 - Rebase onto master
 - apply astyle

**Comment by mchindri** [comment]
v3:
 - Fix cppcheck
 - Keep only 3 values for iio_chan_type

**Comment by mchindri** [comment]
v4:
 - add missing ;

File: libraries/iio/iio.c [hunk 1]
```diff
@@ -742,6 +696,103 @@ ssize_t iio_step(struct iio_desc *desc)
...
+		n = 0;
+	else
+		n = buff_size;
+
+	if (buff == NULL)
+		/* Set dummy value for buff. It is used only for counting */
+		buff = ch_id;
+
+	i = 0;
+	i += snprintf(buff, max(n - i, 0),
```
Comment (buha) [inline libraries/iio/iio.c:741]: snprintf may return a negative number and you assume all returns are positive

Comment (mchindri) [reply libraries/iio/iio.c:741]: I researched a bit and from the documentation, negative error will be returned when a **encoding ** error occurs and encoding only occurs when using unicode strings with "%ls". I use only ascii string so no encoding happens.
I tested with all ascii characters and no error.

**Comment by mchindri** [comment]
will merge today if no more comments

**Pull request merged**